hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMAllocateBlockRequest.java

```diff
@@ -169,7 +173,22 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     OmKeyInfo omKeyInfo = null;
     try {
       // check Acl
-      checkBucketAcls(ozoneManager, volumeName, bucketName, keyName);
+      // Native authorizer requires client id as part of keyname to check
+      // write ACL on key. Add client id to key name if ozone native
+      // authorizer is configured.
+      Configuration config = ozoneManager.getConfiguration();
+      if (OmUtils.isNativeAuthorizerEnabled(config)) {
```

Comment:
I think here, instead of reading this from config every time for each operation, as this is in hot path, we can make add an method in OzoneManager, to return true/false depending on Authorizer.

https://github.com/apache/hadoop/blob/trunk/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L474
We can declare a variable boolean isNativeAuthorizer in OzoneManager. And set to true. And then we can expose a method to return this variable value through isNativeAuthorizer()